### PR TITLE
initial conversion of main services - build and deploy

### PR DESCRIPTION
**Pull request by matthewgardner**
THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE FINOS Corporate Contributor License Agreement.

THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

https://github.com/finos/traderX/issues/203

Complete
- Added Tilt file for local execution
- Added Base k8s config
- Added Yaml for most services
- Simple Port Ingress (will revisit in another PR)

### CODE REVIEW

**Comment by DovOps** [comment]
@matthewgardner  once we sort out the CLA situation, we're ready to merge this. I think you can do this by clicking the red CLA button above.

**Comment by matthewgardner** [comment]
- Added Nginx Ingress
- Added to readme for how to add to local cluster

Screenshot 2024-09-10 at 21 02 29

Suggestions for future pr (out of scope of this PR?) is to enable health endpoints on service so make it easy to spot when services are actually ready.

File: web-front-end/angular/Dockerfile [hunk 1]
```diff
@@ -17,4 +17,4 @@ RUN su node -c "source /usr/local/share/nvm/nvm.sh && nvm install ${EXTRA_NODE_V
 # [Optional] Uncomment if you want to install more global node modules
 # RUN su node -c "npm install -g <your-package-list-here>"
 RUN npm install --ignore-scripts puppeteer
-ENTRYPOINT npm run start-prod
\ No newline at end of file
+ENTRYPOINT npm run ${WEB_FRONT_END_START_MODE:-start-prod}
```
Comment (matthewgardner) [inline web-front-end/angular/Dockerfile:20]: @DovOps  should i revert this change?

Comment (DovOps) [reply web-front-end/angular/Dockerfile:20]: Yeah, if ingress is working you don't need to connect to each service via its port, but rather connect only via the ingress controller, but test to make sure the app works fine that way.

**Comment by DovOps** [comment]
> 
> Suggestions for future pr (out of scope of this PR?) is to enable health endpoints on service so make it easy to spot when services are actually ready.

Definitely. See issue #124

File: web-front-end/angular/Dockerfile (hunk 1 above)
Comment (matthewgardner) [reply web-front-end/angular/Dockerfile]: Done - also tested on a windows env :)

**Comment by DovOps** [comment]
@bingenito  @maoo  @karlmoll  - Need a bit of help sorting out the easyCLA issue, as it has to do with a commit from me that is not using the right e-mail.  Tried fixing this but it did not work. Once done we can merge this, unless we should merge and note the exception.

**Comment by maoo** [comment]
> @bingenito @maoo @karlmoll - Need a bit of help sorting out the easyCLA issue, as it has to do with a commit from me that is not using the right e-mail. Tried fixing this but it did not work. Once done we can merge this, unless we should merge and note the exception.

@DovOps , could you please confirm that the email address that you see on https://api.github.com/repos/finos/traderX/commits/cd9ffec6febfe75913dcf95e26e0b1682e75c79f is also on https://github.com/settings/emails ?

In parallel, I'd also check `git config --list | grep email` to make sure which email you use to sign commits; another - maybe longer - option is to update the `user.email` config, `git reset` and then commit again on @matthewgardner branch.

**Comment by DovOps** [comment]
@maoo  it's the noreply one for me, which apparently was in my settings on the local repo for some reason.  I'll try a git reset + commit if that works

**Comment by DovOps** [comment]
This did not succeed. Merging this as-is.  @bingenito  can confirm that @DovOps  (author of https://github.com/finos/traderX/commit/cd9ffec6febfe75913dcf95e26e0b1682e75c79f ) has agreed to CLA.

**Comment by bingenito** [comment]
> This did not succeed. Merging this as-is. @bingenito can confirm that @DovOps (author of [cd9ffec](https://github.com/finos/traderX/commit/cd9ffec6febfe75913dcf95e26e0b1682e75c79f) ) has agreed to CLA.

As CLA Manager for Morgan Stanley I confirm that @DovOps is covered under our EasyCLA
